sdk/src/main/scala/com/horizen/account/websocket/WebSocketAccountServerEndpoint.scala

```diff
@@ -0,0 +1,308 @@
+        case UNSUBSCRIBE_REQUEST.request =>
+          unsubscribe(session, rpcRequest)
+        case unknownMethod =>
+          WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+            new RpcError(RpcCode.MethodNotFound, s"Method $unknownMethod not supported.", "")),
+            session)
+          log.debug(s"Method $unknownMethod not supported.")
+      }
+    } catch {
+      case ex: Throwable =>
+        val sw = new StringWriter
+        ex.printStackTrace(new PrintWriter(sw))
+        WebSocketAccountServerEndpoint.send(new RpcResponseError(new RpcId("1"),
+          new RpcError(RpcCode.ExecutionError, "Websocket On receive message processing exception occurred", sw.toString)),
+          session)
+        log.debug("Websocket On receive message processing exception occurred = " + ex.getMessage)
+    }
+  }
+
+  private def subscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
+    val subscribeMethod = rpcParams.get(0)
+    val subscriptionId = createSubscriptionId()
+
+    subscribeMethod.asText() match {
+
+      case NEW_HEADS_SUBSCRIPTION.method =>
+        WebSocketAccountServerEndpoint.addNewHeadsSubscription(Subscription(session, subscriptionId))
+        log.debug("New Subscription on newHeads "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case NEW_PENDING_TRANSACTIONS_SUBSCRIPTION.method =>
+        WebSocketAccountServerEndpoint.addNewPendingTransactionsSubscription(Subscription(session, subscriptionId))
+        log.debug("New Subscription on newPendingTransactions "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case LOGS_SUBSCRIPTION.method =>
+        if (rpcParams.size() < 2)
+          WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+            new RpcError(RpcCode.InvalidParams, "Missing filters (address, topcis).", "")),
+            session)
+        val logs = rpcParams.get(1)
+        val addressFilter = getLogAddresses(logs)
+        val topicFilter = getLogTopics(logs)
+        WebSocketAccountServerEndpoint.addLogsSubscription(SubscriptionWithFilter(session, subscriptionId,
+          addressFilter,
+          topicFilter
+        ))
+        log.debug("New Subscription on logs "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case uknownMethod =>
+        WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+          new RpcError(RpcCode.InvalidParams, "unsupported subscription type " + uknownMethod, "")),
+          session)
+        log.debug("unsupported subscription type " + uknownMethod)
+    }
+
+  }
+
+  private def unsubscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
```

Comment:
validate the data first.
1. Do we accept multiple ids?
2. In eth examples I see subscription id to be 16 bytes, is it important? any limit?
3. in case subscription note found, why do we return error, but not `false`?

Reply:
1. Yes
2. The length is not important, inside Ganache the id is a progressive counter and I thought it could be a good idea
3. This is the Ethereum behavior but we can change it if you want